src/lib/utils/FormatDate.android.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+import { filter } from 'lodash'
+import moment from 'moment'
+import 'moment/min/locales'
+
+import { locale } from './i18n.js'
+
+const shortDateTimeFormat = (() => {
+  // do not change locale globally just for particular instance
+  const now = moment().locale(locale)
+  const localeData = now.localeData()
+
+  // detect date format
+  const dateFormat = localeData.longDateFormat('L')
+
+  return `${dateFormat} HH:mm`
+})()
+
+/**
+ * format date util function
+ * @param {string} stringDate
+ */
+export const getFormattedDateTime = (stringDate: string = null) =>
+  moment(...filter([stringDate])).format(shortDateTimeFormat)
```

Comment:
has this been tested?
i dont think it will work as you dont set the locale on the actual moment instance, nor set it globally
move everything inside shortDateTimeFormat which should return the actual date to display

Reply:
1. we detecting locale once (defined in i18n.js)
2. we detecting proper date format once in the IIFE lines 8-17. 
2.1. on line 10 we're setting locale to moment instance then getting locale options
2.2. once we've got the proper date format, we're completing it with HH:mm and returning
2.3 the result this IIFE returns is assigned to shortDateTimeFormat
3. getFormattedDateTime uses shortDateTimeFormat which is already initialized by IIFE and contains the proper datetime format

Reply:
also, to answer the original question - yes, it has been tested